src/tasks/MobileBuildTasks/Android/Ndk/NdkTools.cs

```diff
@@ -0,0 +1,117 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.Android.Build.Ndk
+{
+    public sealed class NdkTools
+    {
+        private string toolRootPath;
+        private string toolPrefixPath;
+        private string asPrefixPath;
+
+        private string clangPath;
+        private string ldName;
+        private string ldPath;
+
+        private string netArch;
+        private string hostOS;
+        private string apiLevel;
+
+        private static readonly Dictionary<string, string> validHosts = new Dictionary<string, string>()
+        {
+            { "osx", "darwin-x86_64" },
+            { "linux", "linux-x86_64" },
+            { "windows", "windows-x86_64" }
+        };
+
+        private static readonly Dictionary<string, AndroidArch> validArches = new Dictionary<string, AndroidArch>()
+        {
+            { "arm", new AndroidArch("32-bit ARMv7", "arm", "armeabi-v7a", "arm-linux-androideabi") },
+            { "arm64", new AndroidArch("64-bit ARMv8", "aarch64", "aarch64-v8a", "aarch64-linux-android") },
+            { "x86", new AndroidArch("32-bit Intel", "x86", "x86", "i686-linux-android") },
+            { "x64", new AndroidArch("64-bit Intel", "x86_64", "x86_64", "x86_64-linux-android") }
+        };
+
+        public NdkTools(string netArch, string hostOS, string apiLevel)
+        {
+            string cmdExt = Utils.IsWindows() ? ".cmd" : "";
+
+            this.netArch = netArch;
+            this.apiLevel = apiLevel;
+
+            ValidateRequiredProps(hostOS);
+
+            this.hostOS = validHosts[hostOS];
+
+            toolRootPath = Path.Combine(Ndk.NdkPath, "toolchains", "llvm", "prebuilt", this.hostOS);
+
+            asPrefixPath = Path.Combine(toolRootPath, Triple, "bin");
+            toolPrefixPath = Path.Combine(toolRootPath, "bin");
+
+            clangPath = Path.Combine(ToolPrefixPath, $"{Triple}{apiLevel}-clang{cmdExt}");
+            ldPath = toolPrefixPath;
+            ldName = "ld";
+        }
+
+        public string ToolPrefixPath
+        {
+            get => toolPrefixPath;
+        }
+
+        public string AsPrefixPath
+        {
+            get => asPrefixPath;
+        }
+
+        public string Triple
+        {
+            get => validArches[netArch].Triple;
+        }
+
+        public string LdName
+        {
+            get => ldName;
+        }
+
+        public string LdPath
+        {
+            get => ldPath;
+        }
+
+        public string ClangPath
+        {
+            get => clangPath;
+        }
+
+        private void ValidateRequiredProps(string hostOS)
+        {
+            if (Ndk.NdkVersion.Main.Major != 23)
+            {
+                throw new Exception($"NDK 23 is required. An unsupported NDK version was found ({Ndk.NdkVersion.Main.Major}).");
```

Comment:
is it really the exact version? not >= ?

Reply:
Yes, for now.  I'll follow up with more broad support.